src/js/lib/client/tasks/signArchives.js

```diff
@@ -0,0 +1,256 @@
+/**
+ This script signs the content of all archives that we have for macOS. For this to work this needs
+ to run on macOS with `codesign` and a JDK installed. `codesign` is needed to sign the files,
+ while the JDK is needed for correct packing and unpacking of java archives.
+
+ Our use case requires us to re-sign jar contents that cannot be opened as pure
+ zip archives, but require a java toolchain to extract and re-assemble.
+ This code is based on https://github.com/electron/electron-osx-sign/pull/231 but our use case
+ is unlikely to be supported by electron-osx-sign as adds a java toolchain as additional
+ dependency.
+ This script should be removed once the engine is signed.
+**/
+const path = require('path')
+const child_process = require('child_process')
+const { dist } = require('../../../../../build/paths')
+
+const contentRoot = path.join(dist.root, 'client', 'mac', 'Enso.app', 'Contents')
+const resRoot = path.join(contentRoot, 'Resources')
+
+// TODO: Remove this once https://github.com/enso-org/ide/issues/1359 has been implemented.
+const ENGINE = '0.2.11'
+const ID = '"Developer ID Application: New Byte Order Sp. z o. o. (NM77WTZJFQ)"'
+// Placeholder name for temporary archives.
+const tmpArchive = 'compressed.zip'
```

Comment:
I think this name is confusing. `.zip` tells it is compressed. The name should tell whats inside. Naming a zip file compressed is like nameing a variable "var" to me :P